_books/ion-1-1/src/macros/system_macros.md

```diff
@@ -0,0 +1,417 @@
+## System Macros
+
+Many of the system macros _MAY_ be defined as template macros, and when possible, the specification includes a template.
+Templates are given here as normative example, but system macros are not required to be implemented as template macros.
+
+The macros that can be defined as templates are included as system macros because of their broad applicability, and
+so that Ion implementations can provide optimizations for these macros that run directly in the implementations runtime
+environment rather than in the macro evaluator.
+For example, a macro such as [`add_symbols`](#add_symbols) does not produce user values, so an Ion Reader could bypass
+evaluating the template and directly update the encoding context with the new symbols.
+
+### Stream Constructors
+
+#### `none`
+
+```ion
+(macro none () (.values))
+```
+
+`none` accepts no values and produces nothing (an empty stream).
+
+#### `values`
+
+```ion
+(macro values (v*) v)
+```
+
+This is, essentially, the identity function. 
+It produces a stream from any number of arguments, concatenating the streams produced by the nested expressions.
+Used to aggregate multiple values or sub-streams to pass to a single argument, or to produce multiple results.
+
+#### `flatten`
+
+```ion
+(macro flatten (sequence*) /* Not representable in TDL */)
+```
+The `flatten` system macro constructs a stream from the content of one or more sequences.
+
+Produces a stream with the contents of all the `sequence` values.
+Any annotations on the `sequence` values are discarded.
+Any non-sequence arguments will raise an error.
+Any null arguments will be ignored.
+
+Examples:
+```ion
+(:flatten [a, b, c] (d e f)) => a b c d e f
+(:flatten [[], null.list] foo::()) => [] null.list
+```
+
+The `flatten` macro can also be used to splice the content of one list or s-expression into another list or s-expression.
+```ion
+[1, 2, (:flatten [a, b]), 3, 4] => [1, 2, a, b, 3, 4]
+```
+
+#### `parse_ion`
+
+Ion documents may be embedded in other Ion documents using the `parse_ion` macro.
+
+```ion
+(macro parse_ion (uint8::data*) /* Not representable in TDL */)
```

Comment:
I suggest using a tagged encoding for `data`; I think using `uint8` would cause tagless encoding domain constraints to apply, making this weird for text inputs.

For binary it looks pretty nice though!
```
(.parse_ion 
  (..
    0xE0 0x01 0x01 0xEA    // $ion_1_1
    0xA3 0x66 0x6F 0x6F    // foo
  )
)
```

This does make me want hex blob syntax 😢

Reply:
Thanks for catching that. That was an idea I was toying with, but I didn't intend for that to make it into the PR.